designs/odcr.md

```diff
@@ -0,0 +1,213 @@
+`p5.48xlarge` or other large GPU instance types.
+
+### Capacity Reservations
+
+Each [Capacity Reservation](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2@v1.162.1/types#CapacityReservation) is defined with:
+
+- The Availability Zone in which to reserve the capacity
+- The number of instances for which to reserve capacity
+- The instance attributes, including the instance type, tenancy, and platform/OS
+- Instance match criteria
+  - Targeted -- only accept instances that matches all attributes + explicitly targeted the capacity reservation
+  - Open -- if capacity reservation accepts all instances that matches all attributes
+- A start and end date (if applicable) for when the reservation of capacity is available
+
+AWS also supports grouping Capacity Reservation into Capacity Reservation groups. 
+Both these entities are supported in Launch Template's CapacityReservationTarget [definitions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-capacityreservationtarget.html).
+
+## Goals
+
+- Support associating ODCRs to EC2NodeClass
+  - open
+  - targeted
+- Define Karpenter's behavior when configuring Capacity Reservations for an EC2NodeClass
+- Define Karpenter's behavior when encountering errors when attempting to launch nodes into Capacity Reservation
+- Define Karpenter's behavior when consolidating nodes to Capacity Reservation when available
+
+## Non-Goals
+
+- Supporting capacity-blocks as a capacity-type.
+- Supporting changes in scaling behavior when ODCR is associated to a NodeClass. _We won't bring up N nodes to match an N node capacity reservation, this would directly interfear with the ability to share a Capacity Reservation between multiple clusters or accounts. The first Karpenter finding the Capacity Reservation would provision all instances for a reservation leaving nothing unused_
+- Supporting Capacity Reservation Groups. _Adding this abstraction for now adds additional complexity_
+- Supporting updating Capacity Reservations availabile instance count after success of CreateFleet API. _We will currenlty rely on reconsolidation of capacity reservations itself, and rely on CreateFleet API throwing a `ReservationCapacityExceeded`_
+- Supporting conditions for capacity reservation instance creations. _By showing the available instance count within each capacity reservation for an EC2NodeClass we have similar information available_
+
+## Proposed Solution
+
+### Supporting associating Capacity Reservations to EC2NodeClass
+
+- Add a new field under `spec` for `capacityReservationSelectorTerms` to `EC2NodeClass` for defining which Capacity Reservation to be used for a specific `EC2NodeClass`
+  - This will allow us to attach multiple Capacity Reservations across AZs and Instance Types to a single EC2NodeClass. This capability removes the need for Capacity Reservation Groups for this MVP.
```

Comment:
I'm not sure that it "removes the need", but I get what you mean -- it's basically doing the same thing that CRG is doing, though, I think it may be worth calling out that allowing CRG selection is probably still necessary to reduce the migration burden on any users that are going to be onboarding to ODCRs with Karpenter